Control/Monad/Accum.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE CPP #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Control.Monad.Accum
+-- Copyright   :  (c) Manuel Bärenz,
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  libraries@haskell.org
+-- Stability   :  experimental
+-- Portability :  non-portable (multi-param classes, functional dependencies)
+--
+-- The MonadAccum class.
+--
+--      Inspired by the paper
+--      /Functional Programming with Overloading and Higher-Order Polymorphism/,
+--        Mark P Jones (<http://web.cecs.pdx.edu/~mpj/pubs/springschool.html>)
+--          Advanced School of Functional Programming, 1995.
+-----------------------------------------------------------------------------
+
+module Control.Monad.Accum (
```

Comment:
Instead of re-exporting the module, we need the haddocks to look like others. In order to do that, we need three sections:
* MonadAccum class
* The Accum monad
* The AccumT monad transformer

Please refer to eg the documentation for Control.Monad.Reader